api/poetry.lock

```diff
@@ -8238,20 +8441,19 @@ files = [
 
 [[package]]
 name = "urllib3"
-version = "2.0.7"
+version = "1.26.19"
```

Comment:
It's a surprising downgrade in `urllib3` caused and required by `alibabacloud-tea` .

Reply:
Indeed, it's quite strange. The `alibabacloud-tea` package requires a version of `urllib3` that should be `urllib3<3,>=1.21.1`. I don't quite understand why Poetry downgraded the version to 1.26.19.

Reply:
![image](https://github.com/langgenius/dify/assets/1935105/a261f6e7-45cd-4ae6-8c60-66baa9ec7055)
From the poetry lockfile, `alibabacloud-tea` does enforce to require urllib3 1.x.

Reply:
How about use the latest minor version first, like `alibabacloud_tea_openapi="~0.3.10"`

Reply:
I have identified the source of the error, which is due to an inappropriate `egg-info/requires.txt` from the `alibabacloud-tea` dependency that `alibabacloud_tea_openapi` relies on. The `setup.py` for the `alibabacloud-tea` library checks the OpenSSL version and adds the requirement for `urllib3<2.0.0` if OpenSSL is missing or the version is too low. You can see this [here](https://github.com/aliyun/tea-python/blob/8073e26f8523935e068eb25e5c16e7236ca92608/setup.py#L41).   
Unfortunately, it seems that the machine used for packaging and uploading to PyPI did not have OpenSSL installed, causing this requirement to be unconditionally written into the `egg-info/requires.txt`, which was then downloaded and used by Poetry for dependency analysis.   
I have reached out to the developers of `alibabacloud-tea`, and they have indicated that the issue may be corrected in the next couple of days.   
Please check back here later. Thank you for your patience.

Reply:
I'm okay with it, if this downgrade is necessary. It would be just nice to be clarified in description or somewhere else to show the major changes of this PR.

Reply:
Btw, what does the word `tea` inside the package name stand for ? I am green to Aliyun or ADB's service.

Reply:
err...I'm not sure, but it seems to stand for "Tea API Gateway", a tool to work with alibabacloud api, defined in [alibabacloud-apigateway-core-sdk](https://github.com/aliyun/alibabacloud-apigateway-core-sdk)

Reply:
Hard to understand , but still good to know then. Thanks, haha.